llpc/util/llpcCacheAccessor.h

```diff
@@ -51,6 +51,36 @@ class CacheAccessor {
     initializeUsingBuildInfo(buildInfo, cacheHash, compiler);
   }
 
+  CacheAccessor(CacheAccessor &&ca) { *this = std::move(ca); }
+
+  CacheAccessor &operator=(CacheAccessor &&ca) {
```

Comment:
Why not use a default one? I would not expect anything to use/inspect moved-from objects and depend on their internal state. This is kind of like setting variables in destructor.

Reply:
Just because we do not expect people to use a moved-from object that does not mean it will not happen.  The object is not destroyed, and could potentially be accessed.  I want to keep it in a known state that will not allows a read from a potentially invalid pointer.

Also, the destructor will still be called so the state has to be reasonable.

Reply:
The only two operations that make sense on moved-from objects are re-assignment and destruction.

>  I want to keep it in a known state that will not allows a read from a potentially invalid pointer.
By the same logic, you would have to zero-out all pointers after in the destructor.

While it makes sense to me to leave objects in neat state, this seems to me like an unnecessary burden. Typically, I only expect to see hand-written move constructors for objects that manage some raw resources, and seeing a constructor like this makes me double-check if I'm missing something about that class. Next, you have to remember to  to update the constructor every time you add new fields. If you miss that, you may end up not moving a new field and create a bug that would have been prevented by using a default move constructor.

I don't want to block this review over this, but I'm not a fan of this implementation.

Reply:
Ah, I missed that the destructor does something special to the `BinaryData` field. This is the piece that I was missing. Could you add a comment that the destructor accesses this POD field so that we have to set it to the default value?